internal/cache/backend.go

```diff
@@ -149,7 +157,19 @@ func (b *Backend) loadFromCacheOrDelegate(ctx context.Context, h restic.Handle,
 }
 
 // Load loads a file from the cache or the backend.
+// For loading data files, please use LoadData!
```

Comment:
I'm not sure if this really simplifies things. The cached backend still implements Backend, but is no longer supposed to be used that way. The comment isn't clear on the relation between the two: if client code does use Load, will it break?